src/main/java/io/quarkus/security/identity/SecurityIdentity.java

```diff
@@ -133,4 +139,30 @@ default <T extends Principal> T getPrincipal(Class<T> clazz) {
     default boolean checkPermissionBlocking(Permission permission) {
         return checkPermission(permission).await().indefinitely();
     }
+
+    /**
+     * Checks if a user holds a given permission.
+     * <p>
+     * This method is asynchronous, as it may involve calls to a remote resource.
+     *
+     * @param permission The permission
+     * @return A completion stage that will resolve to true if the user has the specified permission
```

Comment:
I am not sure how much `Uni` is a `completion stage`. I can be wrong though. I found https://smallrye.io/smallrye-mutiny/latest/guides/completion-stage/#from-uni-to-completionstage but that speaks about the `CompletionStage` type. Is it really correct?

Reply:
@michalvavrik It is just a copy and paste from some other method, OK, I see it it is the the already existing  `boolean checkPermissionBlocking(Permission permission)` java Docs, so that it needs to be fixed

Reply:
I can handle Java docs update independently, we don't have to rush with this PR if there are still concerns